filterAPI/post_filter_blueprint_test.go

```diff
@@ -18,33 +19,55 @@ func TestSuccessfullyPostFilterBlueprintForPublishedInstance(t *testing.T) {
 	instanceID := uuid.NewV4().String()
 	dimensionOptionOneID := uuid.NewV4().String()
 	dimensionOptionTwoID := uuid.NewV4().String()
+	datasetID := uuid.NewV4().String()
+	editionID := uuid.NewV4().String()
+	edition := "2017"
+	version := 1
 
 	filterAPI := httpexpect.New(t, cfg.FilterAPIURL)
 
 	var docs []*mongo.Doc
 
+	dataset := &mongo.Doc{
+		Database:   cfg.MongoDB,
+		Collection: "datasets",
+		Key:        "_id",
+		Value:      datasetID,
+		Update:     datasetJSON.ValidPublishedWithUpdatesDatasetData(datasetID),
+	}
+
+	editionDoc := &mongo.Doc{
+		Database:   cfg.MongoDB,
+		Collection: "editions",
+		Key:        "_id",
+		Value:      datasetID,
+		Update:     datasetJSON.ValidPublishedEditionData(datasetID, editionID, edition),
+	}
+
 	instance := &mongo.Doc{
 		Database:   cfg.MongoDB,
 		Collection: "instances",
 		Key:        "instance_id",
 		Value:      instanceID,
-		Update:     GetValidPublishedInstanceDataBSON(instanceID),
+		Update:     GetValidPublishedInstanceDataBSON(instanceID, datasetID, edition, version),
```

Comment:
do we still need instance ID in this parameter list? consider removing it possible

Reply:
Not for our current tests, but if we needed two versions within one test then instanceID would be needed, to save duplicating the doc twice in json.go - so I think i will leave this in here for now.